fastparquet/api.py

```diff
@@ -1263,3 +1415,45 @@ def row_groups_map(rgs: list) -> dict:
         file = rg.columns[0].file_path
         files_rgs[file].append(rg)
     return files_rgs
+
+
+def partitions(row_group, only_values=False) -> str:
+    """Returns partition values as string.
+    
+    Values of partitions are separated with '/'.
+
+    Parameters
+    ----------
+    row_group : obj or str
+        Row group object or row group `file_path` as given by
+        `rg.columns[0].file_path`.
+    only_values: bool, default False
+        If False, only values of partitions are returned;
+        If True, names and values of partitions are returned (faster).
+
+    Returns
+    -------
+    str
+        Paritions values.
+    """
+    f_path = (row_group if isinstance(row_group, str)
+              else row_group.columns[0].file_path)
+    if '/' in f_path:
+        return ('/'.join(re.split('/|=', f_path)[1::2]) if only_values
+                else f_path.rsplit('/',1)[0])
+
+
+def part_ids(row_groups) -> dict:
```

Comment:
It is based on `find_max_part()`, but return the list of partition files as a dict with:
- keys as partition file ID
- values as tuples `row_group_id, part_name`.
All this information are re-used in `ParquetFile._sort_part_names()`.
It is also used for `find_max_part` which apply a `max()` on the keys.
(I wanted to remove `find_max_part` but it is directly called in a dask test case)

> Also, I note it will fail if any filenames do not match the given pattern.

You are right, I corrected the code as it was in `find_max_part` to have it not failing if  pattern is not matched.